lib/routes/v1.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+// this router dynamically handle model logic based on paramter attahed to the request
+
+const express = require('express');
+const router = express.Router();
+
+// bring in MOdels
+const ProductModel = require('../models/products/products-model.js');
+const CategoryModel = require('../models/categories/categories-model.js');
+
+//define some paramter middleware
+//we can look for a specific param in a route, and modify the request object
+// In this way, we can communiate with handlers that are next in the request 'list'
+
+function getModel(req, res, next) {
+  // console.log(req);
+  let model = req.params.model;
+  switch (model) {
+    case 'categories':
+      req.model = new CategoryModel();
+      next();
+      break;
+    case 'products':
+      req.model = new ProductModel();
+      next();
+      break;
+    default:
+      next('Invalid Model');
+      break;
+  }
+}
+
```

Comment:
I am getting an error trying to GET to categories or products. I think that you need to bring in your `mongo-interface.js` then when you instanciate you would feed it the schema for categories or products. Something like:

`req.model = new MongoInterface(CategoriesSchema);`